VisualizationBase/src/nodes/ViewItemNode.cpp

```diff
@@ -62,4 +64,73 @@ ViewItemNode* ViewItemNode::withReference(Model::Node *reference, int purpose)
 	return result;
 }
 
+ViewItemNode* ViewItemNode::withJson(QJsonObject json, const ViewItem *parent)
```

Comment:
I would call this method `fromJson` and leave it static. Then put all the stuff that's currently in the other `fromJson` method here.